docs/proposals/multi-arch-image-mgmt.md

```diff
@@ -0,0 +1,328 @@
+| v1-linux-arm64  |    sha:256@zzzz          |     linux/arm64   |
+
+
+## Investigation on other client tools and industry
+
+Most of popular container client tools support create and push a multi-arch image using Docker manifest list or OCI image index format, but these tools **require users to push platform-specific image to the registry separately**. They don’t provide native support for local build experience.
+
+- **Docker**:
+  - [Docker buildx buildx](https://docs.docker.com/reference/cli/docker/buildx/build/)
+  - [Docker buildx imagetool](https://docs.docker.com/reference/cli/docker/buildx/imagetools/)
+  - [Docker manifest create](https://docs.docker.com/reference/cli/docker/manifest/)
+- **Podman (Backed by Red Hat)**: similar with “docker manifest”, it provides _“podman manifest”_ with subcommands to create and manipulate manifest lists and image indexes
+- Crane **(Backed by Google)**: provides a single command [_“crane index append”_](https://github.com/google/go-containerregistry/blob/main/cmd/crane/recipes.md#create-a-multi-platform-image-from-scratch) to composite an image index
+- **Regctl (Individual)** provides [_“regctl index add/create/delete”_](https://github.com/regclient/regclient/blob/main/docs/regctl.md#index-commands) to creates or manages OCI Indexes and Manifest Lists
+- **manifest-tool (Individual from AWS):** create Docker manifest list or OCI image index in a registry by using the [**_manifest-tool_** **_push_** command with either a YAML file describing the images to assemble or by using a series of parameters](https://github.com/estesp/manifest-tool?tab=readme-ov-file#sample-usage).
+- **Skopeo (Backed by Red Hat)**: doesn’t support OCI image index and Docker manifest list
+
+Based on the industry observation, Docker buildx plugin supports building a multi-arch image using the OCI image index format. Even [Docker Official Images](https://docs.docker.com/trusted-content/official-images/) are using the OCI image index format to create multi-platform images.
+
+[Homebrew](https://github.com/orgs/Homebrew/packages) publishes all images on ghcr.io using OCI image index. Docker provides two ways to build a multi-arch image, i.e. _“_[_docker manifest_](https://docs.docker.com/reference/cli/docker/manifest/)_”_ and _“_[_docker buildx_](https://docs.docker.com/reference/cli/docker/buildx/build/)_”._ However, there are some limitations and problems of using Docker to build multi-arch images in certain scenarios and environments.
+
+## Open Questions
+
+- Is multi-arch artifact in the scope of the `oras manifest` subcommands? Is the multi-arch artifact popular in the industry?
+- Should ORAS support creating Docker manifest list?
+- Is ORAS supposed to add flags “--artifact-type” and `--subject` to `oras manifest index create`?
+- Is it much extensible to introduce a new command `oras annotate` or `oras manifest index annotate` to add or modify annotations of an image index with its child image?
```

Comment:
We'd better discuss `oras annotate` against `oras attach --annotation`. Meanwhile, `oras manifest index annotate` is not a good one since it only scopes to the index and does not cover individual manifest.

Reply:
This will be follow-up enhancements in the next release.

See the issue: Support attach annotations to image index and its child image manifest in `oras attach`: https://github.com/oras-project/oras/issues/1531